scripts/run_lighthouse_tests.py

```diff
@@ -50,11 +53,94 @@ def run_lighthouse_checks():
+    app_yaml_filepath = 'app.yaml'
+
+    python_utils.PRINT('Starting oppia server...')
+
+    p = subprocess.Popen(
+        '%s %s/dev_appserver.py --host 0.0.0.0 --port %s '
+        '--clear_datastore=yes --dev_appserver_log_level=critical '
+        '--log_level=critical --skip_sdk_update_check=true %s' % (
+            common.CURRENT_PYTHON_BIN, common.GOOGLE_APP_ENGINE_HOME,
+            APP_ENGINE_PORT, app_yaml_filepath), shell=True)
+    RUNNING_PROCESSES.append(p)
+
+
+def download_and_install_nginx():
+    """Download and install nginx."""
+    python_utils.PRINT('Installing nginx...')
+    update_command = ['sudo', 'apt', 'update']
+    install_command = ['sudo', 'apt', 'install', 'nginx']
+
+    subprocess.check_call(update_command)
+    subprocess.check_call(install_command)
+
+
+def start_proxy_server():
+    """Start the nginx proxy server."""
+    filepath = os.path.join(common.CURR_DIR, 'nginx.conf')
+
+    python_utils.PRINT('Starting proxy server...')
+    start_server_command = ['sudo', 'nginx', '-c', filepath]
+    subprocess.Popen(start_server_command)
+
+
+def run_lighthouse_checks_with_compression():
+    """Run lighthouse checks with compression enabled."""
+    try:
+        python_utils.PRINT('Checking if nginx is installed...')
+        check_nginx_command = ['which', 'nginx']
+        subprocess.check_call(check_nginx_command)
+    except subprocess.CalledProcessError:
+        download_and_install_nginx()
+
+    constants_env_variable = '"DEV_MODE": false'
+    for line in fileinput.input(
+            files=[os.path.join('assets', 'constants.ts')], inplace=True):
+        # Inside this loop the STDOUT will be redirected to the file,
+        # constants.ts. The end='' is needed to avoid double line breaks.
+        python_utils.PRINT(
+            re.sub(
+                r'"DEV_MODE": .*', constants_env_variable, line), end='')
+
+    start_google_app_engine_server()
+    common.wait_for_port_to_be_open(APP_ENGINE_PORT)
+    start_proxy_server()
+    common.wait_for_port_to_be_open(NGINX_PORT)
     run_lighthouse_checks()
-    delete_reports()
+
+
+def cleanup():
+    """Kill the running subprocesses and server fired in this program."""
+    dev_appserver_path = '%s/dev_appserver.py' % common.GOOGLE_APP_ENGINE_HOME
+    processes_to_kill = [
+        '.*%s.*' % re.escape(dev_appserver_path)
+    ]
+
+    for p in RUNNING_PROCESSES:
+        p.kill()
+
+    for p in processes_to_kill:
+        common.kill_processes_based_on_regex(p)
+
+    stop_server_command = ['sudo', 'service', 'nginx', 'stop']
+    subprocess.Popen(stop_server_command)
+
+
+def main(args=None):
+    """Runs lighthouse checks and deletes reports."""
+    parsed_args = _PARSER.parse_args(args=args)
+    atexit.register(cleanup)
+    build.main(args=['--prod_env'])
+
+    if parsed_args.disable_compression:
+        start_google_app_engine_server()
+        common.wait_for_port_to_be_open(APP_ENGINE_PORT)
+        run_lighthouse_checks()
```

Comment:
The above three lines seem to duplicate, can you do whatever required when there is not disable_compression and run the above three lines at the end?